scrapy_autoextract/providers.py

```diff
@@ -0,0 +1,111 @@
+from typing import ClassVar, Type
+
+from autoextract.aio import request_raw
+from autoextract.request import Request as AutoExtractRequest
+from autoextract_poet.page_inputs import (
+    AutoExtractArticleData,
+    AutoExtractProductData,
+)
+from scrapy import Request
+from scrapy.settings import Settings
+from scrapy.statscollectors import StatsCollector
+from scrapy_poet.page_input_providers import (
+    PageObjectInputProvider,
+    register,
+)
+
+
+class QueryLevelError(Exception):
```

Comment:
It would be nice to get the error message isolated. See for example this implementation I have in other code:

```py
class QueryLevelError(Exception):
    def __init__(
        self, url: str, page_type: str, query: List, response: Dict, error_msg: str
    ):
        self.url = url
        self.page_type = page_type
        self.query = query
        self.error_msg = error_msg

    def __str__(self):
        return (
            f"QueryLevelError: url={self.url}, page_type={self.page_type}, "
            f"error={self.error_msg}, query:{self.query}"
        )
```

Probably not all this information is required, but I would keep at least the error and the query.

Reply:
Done, `QueryError` exception was improved.